plugins/modules/foreman_role.py

```diff
@@ -54,6 +54,16 @@
     description: Filters with permissions for this role
     required: false
     type: list
+    elements: dict
+    suboptions:
+      permissions:
+        description: List of permissions
+        required: false
```

Comment:
I kind of think that adding a filter w/o permissions is pointless, but I am not sure if setting `required=True` would make it required *always* or just when filters are defined.

Reply:
It's listed as optional in the apidoc.
But setting it required here still allows the whole list of subitems to be empty.

Reply:
Nobody said the apidoc is always right ;-)

Reply:
Are you saying it should be required?

Reply:
I would prefer that, but not block.

You can create filters with no permissions in the UI too. They are just pointless.

Reply:
Yes, you can still provide an empty list now. Which is as pointless, but at least much harder...